beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -425,3 +427,29 @@ func (f *ForkChoice) CommonAncestorRoot(ctx context.Context, r1 [32]byte, r2 [32
 		}
 	}
 }
+
+// InsertOptimisticChain inserts all nodes corresponding to blocks in the slice
+// `blocks`. This slice must be ordered from parent to child. It includes all
+// blocks **except** the first one. All blocks are assumed to be a strict chain
+// where blocks[i].Parent = blocks[i+1]. Also we assume that the parent of the
```

Comment:
Part of me is scared about these assumptions. What would it take to add actual dumb foul proofs to add in an additional check where `blocks[i].Parent = blocks[i+1]` and/or slot has to be in an incremental order

Reply:
But if any of these assumptions fail the insert function will fail